File: LeetCode/python3/697.py
class Solution:
    def findShortestSubArray(self, nums: List[int]) -> int:
        class Num:
            def __init__(self, i):
                self.count = 1
                self.start = i
                self.end = i

        dict = {}
        max_count = 0
        min_len = sys.maxsize
        for i in range(len(nums)):
            if nums[i] not in dict:
                dict[nums[i]] = Num(i)
            else:
                dict[nums[i]].count += 1
                dict[nums[i]].end = i
            max_count = max(max_count, dict[nums[i]].count)
        for key, value in dict.items():
            if max_count == value.count:
                min_len = min(min_len, value.end - value.start)
        return min_len + 1

File: LeetCode/cpp/697.cpp
class Solution {
public:
    int findShortestSubArray(vector<int> &nums) {
        int n = nums.size(), minLen = INT_MAX;
        map<int, vector<int>> m;
        int size = 0;
        for (int i = 0; i < n; ++i) {
            m[nums[i]].push_back(i);
            if (size < m[nums[i]].size()) {
                size = m[nums[i]].size();
            }
        }
        for (auto iter : m) {
            if (iter.second.size() == size)
                minLen = min(minLen, iter.second[iter.second.size() - 1] - iter.second[0] + 1);
        }
        return minLen;
    }
};